server/world/world.go

```diff
@@ -1449,6 +1566,60 @@ func (w *World) tickEntities(tick int64) {
 	w.entitiesToTick = w.entitiesToTick[:0]
 }
 
+// SetRaining toggles raining depending on the raining argument.
+func (w *World) SetRaining(raining bool, x time.Duration) {
+	w.mu.Lock()
+	w.setRaining(raining, x)
+	w.mu.Unlock()
+}
+
+func (w *World) setRaining(raining bool, x time.Duration) {
```

Comment:
Might want to document this and indicate that, as opposed to SetRaining, setRaining does not lock.